Cargo.toml

```diff
@@ -41,11 +41,15 @@ once_cell = "1.12.0"
 
 [dev-dependencies]
 assert_cmd = "2"
-criterion = "0.3"
+criterion = "0.4"
 libflate = "1"
 maplit = "1.0.1"
 pretty_assertions = "1"
 rand = { version = "0.8", features = ["small_rng"] }
+# Force criterion to pull in regex 1.6 instead of 1.5 during minimal version CI
```

Comment:
I think it'd be useful to include a brief comment about *why* we do this in each instance. Forcing a higher version isn't a goal in and of itself. What problem does each of these forcings solve?

Reply:
Yep. Makes sense. I added some more information in the form of comments.